src/library/scala/Option.scala

```diff
@@ -323,6 +323,43 @@ sealed abstract class Option[+A] extends Product with Serializable {
   final def zip[A1 >: A, B](that: Option[B]): Option[(A1, B)] =
     if (isEmpty || that.isEmpty) None else Some((this.get, that.get))
 
+  /** Converts an Option of pairs into an Option of first elements and an Option of second elements.
+    *
+    *  @tparam A1    the type of the first half of the element pairs
+    *  @tparam A2    the type of the second half of the element pairs
+    *  @param asPair an implicit conversion which asserts that the element type
+    *                of this Option is a pair.
+    *  @return       a pair of Options, containing, respectively, the first and second half
+    *                of each element pair of this Option.
+    */
+  final def unzip[A1, A2](implicit asPair: A => (A1, A2)): (Option[A1], Option[A2]) = {
```

Comment:
should this be `A <:< (A1, A2)` instead? I'm not sure how much I like using a view for this.

Reply:
This is the same signature as other container such as Array.unzip They should be aligned I think.

Reply:
That's true. 

Is there a reason that the other `unzip`s use views rather than `<:<`, I wonder?

Reply:
That will be out of my knowledge. Maybe open a new discussion?

Reply:
Yep, sorry; I was asking in hopes that someone with more knowledge than I would know...